src/rebalancers/libraries/RebalanceOptimizationMath.sol

```diff
@@ -0,0 +1,296 @@
+/**
+ * Created by Pragma Labs
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+pragma solidity 0.8.22;
+
+import { FixedPointMathLib } from "../../../lib/accounts-v2/lib/solmate/src/utils/FixedPointMathLib.sol";
+import { LiquidityAmounts } from "../libraries/uniswap-v3/LiquidityAmounts.sol";
+import { SqrtPriceMath } from "../libraries/uniswap-v3/SqrtPriceMath.sol";
+
+library RebalanceOptimizationMath {
+    using FixedPointMathLib for uint256;
+
+    // The minimal relative difference between liquidity0 and liquidity1, with 18 decimals precision.
+    uint256 internal constant CONVERGENCE_THRESHOLD = 1e6;
+
+    // The maximal number of iterations to find the optimal swap parameters.
+    uint256 internal constant MAX_ITERATIONS = 50;
+
+    event Log(uint256 sqrtPriceNew, uint256 amountIn, uint256 amountOut);
```

Comment:
Remove logs

Reply:
Will do that at the end, need them too often in my tests currently